_source/_posts/2021-07-08-jenkins-ci-dotnet-update-secrets.md

```diff
@@ -0,0 +1,403 @@
+
+In this application, you will create a .NET5 MVC web application and check it into a git repository. You will secure this application with Okta. Finally, you will set up a Jenkins project to fetch the code from your git repository, publish the project, apply configuration values, and run the application. Combining these three technologies you'll see how quick and easy it is to set up a modern CI/CD environment for a web application that is properly secured with a great single sign-on provider.
+
+### What you'll need
+
+> Your favorite IDE. I will be using Visual Studio but Visual Studio Code is another great option
+> A git repository
+> .NET 5
+> Jenkins
+> A Web Browser to Access Jenkins
+> An Okta Developer Account
+> The Okta CLI tool
+
+## Create your Okta Application
+
+The first thing you want to do is create your Okta application. To do this you can use the new [Okta CLI tool](https://cli.okta.com/). Download the Okta CLI tool if you haven't already and run `okta login` to log into your account. Next run `okta apps create` and follow the prompts you receive. First, select `web` as your application type. Next, for your *Redirect URI* type in your web application's development location and append '/okta/callback' to it. For *Post Logout RedirectURI* use your same base domain and append `/signout/callback` as this is the route Okta will use by default. If you have multiple authorization services the CLI tool will prompt you to select one. For this project you can select default however in a production environment you should create a custom server. After a moment the CLI tool will return your *ClientId*. Be sure to note this as you will need it in your application.
+
+Next, navigate to your Okta admin portal. The first thing you want to do is note your *Client Secret*. Next, press **Edit** under the *General Settings* tab and scroll down to the *Login* subsection. Add a value under *Sign-in redirect URIs* with the same path as before but use `https://locahost:5001` instead of your development domain. This URI will be used when you publish and run your application with Jenkins. Similarly, under *Sign-out redirect URIs* add a value with the same route as your development route but using `https://localhost:5001`.  
+
+{% img blog/jenkins-ci-app-secrets-dotnetcore/image1.png alt:"" width:"600" %}{: .center-image }
+
+## Create your .NET Application
+
+Open Visual Studio and select *Create a New Project*. Use the template for *ASP.NET Core Web App(Model-View-Controller)*. Give your application a meaningful name click **Create**. Visual Studio will scaffold your application. Once it is complete you can begin making the necessary changes.
+
+First, install the *Okta.AspNetCore* package from nuget using the command `Install-Package Okta.AspNetCore -Version 3.5.0`.  
```

Comment:
The current text is correct, but it wraps a little funny on our blog, so maybe force the new line (and remove the trailing period).


First, install the *Okta.AspNetCore* package from nuget using the command:

```
Install-Package Okta.AspNetCore -Version 3.5.0
```